feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/proxy/RealProxySyncService.kt

```diff
@@ -0,0 +1,166 @@
+                    LightMetaAccount.Type.POLKADOT_VAULT -> true
+
+                    LightMetaAccount.Type.WATCH_ONLY,
+                    LightMetaAccount.Type.PROXIED -> false
+                }
+            }
+    }
+
+    private suspend fun getSupportedProxyChains(): List<Chain> {
+        return chainRegistry.findChains { it.supportProxy }
+    }
+
+    private suspend fun Chain.getAvailableAccountIds(metaAccounts: List<MetaAccount>): List<MetaAccountId> {
+        return metaAccounts.mapNotNull { metaAccount ->
+            val accountId = metaAccount.accountIdIn(chain = this)
+            accountId?.let {
+                MetaAccountId(accountId, metaAccount.id)
+            }
+        }
+    }
+
+    private suspend fun createMetaAccount(chainId: ChainId, proxiedAccountId: AccountId, identity: Identity?, position: Int): MetaAccountLocal {
+        val chain = chainRegistry.getChain(chainId)
+        return MetaAccountLocal(
+            substratePublicKey = null,
+            substrateCryptoType = null,
+            substrateAccountId = if (chain.isSubstrateBased) proxiedAccountId else null,
+            ethereumPublicKey = null,
+            ethereumAddress = if (chain.isEthereumBased) proxiedAccountId else null,
+            name = identity?.name ?: chain.addressOf(proxiedAccountId),
+            isSelected = false,
+            position = position,
+            type = MetaAccountLocal.Type.PROXIED,
+        )
+    }
+
+    private suspend fun createChainAccount(metaId: Long, chainId: ChainId, accountId: AccountId): ChainAccountLocal {
+        return ChainAccountLocal(
+            metaId = metaId,
+            chainId = chainId,
+            publicKey = null,
+            accountId = accountId,
+            cryptoType = null
+        )
+    }
+
+    private fun createProxyAccount(
+        metaId: Long,
+        chainId: ChainId,
+        proxiedAccountId: AccountId,
+        proxyType: String
+    ): ProxyAccountLocal {
+        return ProxyAccountLocal(
+            metaId = metaId,
+            chainId = chainId,
+            proxiedAccountId = proxiedAccountId,
+            proxyType = proxyType,
+            status = ProxyAccountLocal.Status.ACTIVE
+        )
+    }
+
+    private fun List<ProxiedWithProxies>.formatToLocalProxies(): List<ProxyAccountLocal> {
+        return flatMap { proxied ->
+            proxied.proxies.map { proxy ->
+                createProxyAccount(proxy.metaId, proxied.chainId, proxied.accountId, proxy.proxyType)
+            }
+        }
+    }
+
+    private suspend fun List<ProxyAccountLocal>.loadProxiedIdentities(): Map<ChainId, Map<AccountIdKey, Identity?>> {
+        return this.groupBy { it.chainId }
+            .mapValues { (chainId, proxiesWithProxieds) ->
+                val proxiedAccountIds = proxiesWithProxieds.map { it.proxiedAccountId }
+                identityProvider.identitiesFor(proxiedAccountIds, chainId)
+            }
+    }
+}
```

Comment:
Here I think it is better to not make it an extension since `loadProxiedIdentities` feels more not like an extension for `List<Proxy>` but just a separate method